lib/intervals.js

```diff
@@ -17,34 +18,54 @@ const chunkedOp = require('./chunked-op')
         intervalSize (number): the desired size of each interval in ms
       }
     Returns:
-      {
+      [{
         startDate (number): start of the interval in ms since epoch,
         endDate (number); end of the interval in ms since epoch,
         calories (number): total calories expended during this interval (across all activities)
-      }
+        types (array): normalised activity names during this interval
+      }]
 */
 const calcIntervals = function ({ startDate, endDate, activities, intervalSize }) {
   if (!activities) return []
   let intervals = []
   for (let time = startDate; time < endDate; time += intervalSize) {
     let endTime = time + intervalSize
-    let calories = activities.map(caloriesInPeriod.bind(null, time, endTime)).reduce((x, y) => x + y, 0)
+    let {calories, types} = dataInPeriod(time, endTime, activities)
     if (calories) {
       intervals.push({
         startDate: time,
         endDate: endTime,
-        calories
+        calories,
+        types
       })
     }
   }
   return intervals
 }
 
-function caloriesInPeriod (startDate, endDate, activity) {
+// (num, num, arr) => {calories: 36, types: ['Running']}
+// Type name are ordered with the longest running activity first.
+function dataInPeriod (startDate, endDate, activities) {
+  return activities
+    .map((a) => ({activity: a, ms: activeMillisInPeriod(startDate, endDate, a)}))
+    .sort((a, b) => b.ms - a.ms)
+    .reduce((res, {ms, activity}) => {
+      if (!ms || !activity.calsPerMilli) return res
+      const calories = activity.calsPerMilli * ms
+      // normalise the name of the activity for our intervals
+      const type = activityNames(activity.type)
+      return {
+        calories: res.calories + calories,
+        types: res.types.concat([type])
```

Comment:
Should this be some sort of add-to-set type functionality, to avoid `['Running', 'Running']`?  Google Fit often breaks one actual activity down into multiple chunks.